addons/web/static/tests/views/fields/properties_field_tests.js

```diff
@@ -1861,6 +1862,32 @@ QUnit.module("Fields", (hooks) => {
             await closePopover(target);
             assert.strictEqual(property.querySelector(".o_property_field_value input").value, "");
         }
+
+        // add a new date property
+        await click(target, ".o_field_property_add button");
+        await nextTick();
+        let popover = target.querySelector(".o_property_field_popover");
+        assert.ok(popover, "Should have opened the definition popover");
+        await changeType(target, "date");
+        const type = popover.querySelector(".o_field_property_definition_type input");
+        assert.strictEqual(type.value, "Date", "Should have changed the property type");
+        // choose a default value and check that it is propagated on the property field
+        await click(target, ".o_field_property_definition_value .o_datetime_input");
+        assert.containsOnce(target, ".o_date_picker");
+        await click(getPickerCell("3").at(0));
+        await closePopover(target);
+        const dateRegex = /^[0-9]{2}\/[0-9]{2}\/[0-9]{4}$/g;
```

Comment:
you can call patchDate at the beginning of the test to simulate a given date, then you can assert that you selected a specific date. Typically, we try to freeze external parameters (in this case the current date) s.t. all executions of the test are the same.